File: examples/rand1.py

import random

def begin():
  print "*BEGIN*"
  yield '"output-1"'
  yield '"output-2"'
  yield '"output-3"'

def action(datum):
  yield str(random.random())

def end():
  print "*END*"
  yield "1.23"
  yield "2.34"
  yield "3.45"

File: examples/rand1.R

begin <- function() {
  cat("*BEGIN*")
  emit("\"output-1\"")
  emit("\"output-2\"")
  emit("\"output-3\"")
}

action <- function(datum) {
  emit(toString(runif(1)))
}

end <- function() {
  cat("*END*")
  emit("1.23")
  emit("2.34")
  emit("3.45")
}

